source/code/source/playfab/PlayFabPluginManager.cpp

```diff
@@ -27,47 +27,43 @@ namespace PlayFab
 
     std::shared_ptr<IPlayFabPlugin> PlayFabPluginManager::GetPluginInternal(const PlayFabPluginContract contract, const std::string& instanceName)
     {
-        std::pair<PlayFabPluginContract, std::string> key = std::make_pair(contract, instanceName);
-        const auto pluginEntry = plugins.find(key);
-        if (pluginEntry == plugins.end())
+        PluginEntry& entry = FindOrCreatePluginEntry(contract, instanceName);
+        if (entry.plugin == nullptr)
         {
-            // Requested plugin is not in the cache, create the default one
-            std::shared_ptr<IPlayFabPlugin> pluginPtr = nullptr;
             switch (contract)
             {
             case PlayFabPluginContract::PlayFab_Serializer:
-                pluginPtr = CreatePlayFabSerializerPlugin();
+                entry.plugin = CreatePlayFabSerializerPlugin();
                 break;
             case PlayFabPluginContract::PlayFab_Transport:
-                pluginPtr = CreatePlayFabTransportPlugin();
+                entry.plugin = CreatePlayFabTransportPlugin();
                 break;
             default:
                 throw PlayFabException(PlayFabExceptionCode::PluginAmbiguity, "This contract is not supported");
                 break;
             }
-
-            plugins.insert({ key, pluginPtr });
-            return pluginPtr;
-        }
-        else
-        {
-            return pluginEntry->second;
         }
+        return entry.plugin;
     }
 
     void PlayFabPluginManager::SetPluginInternal(const std::shared_ptr<IPlayFabPlugin>& plugin, const PlayFabPluginContract contract, const std::string& instanceName)
     {
-        std::pair<PlayFabPluginContract, std::string> key = std::make_pair(contract, instanceName);
-        const auto pluginEntry = plugins.find(key);
-        if (pluginEntry == plugins.end())
-        {
-            plugins.insert({ key, std::move(plugin) });
-        }
-        else
+        PluginEntry& entry = FindOrCreatePluginEntry(contract, instanceName);
+        entry.plugin = plugin;
+    }
+
+    PlayFabPluginManager::PluginEntry& PlayFabPluginManager::FindOrCreatePluginEntry(PlayFabPluginContract contract, const std::string& instanceName)
+    {
+        for (PluginEntry& pluginEntry : plugins)
         {
-            plugins.erase(key);
-            plugins.insert({ key, std::move(plugin) });
+            if (pluginEntry.contract == contract &&
+                pluginEntry.name == instanceName)
+            {
+                return pluginEntry;
+            }
         }
+        plugins.push_back({contract, instanceName, nullptr});
+        return plugins.back();
```

Comment:
>back [](start = 23, length = 4)

I didn't realize this in my original proposal, but there's nothing prevent concurrent writes/reads to this vector. I think this function needs to mutually exclude access to this vector to be thread-safe. This problem also existed with the previous map implementation I believe